File: multi-storage-transaction-sample/src/main/java/sample/command/PlaceOrderCommand.java
package sample.command;

import java.util.concurrent.Callable;
import picocli.CommandLine.Command;
import picocli.CommandLine.Parameters;
import sample.Sample;

@Command(name = "PlaceOrder", description = "Place an order")
public class PlaceOrderCommand implements Callable<Integer> {

  @Parameters(index = "0", paramLabel = "CUSTOMER_ID", description = "customer ID")
  private int customerId;

  @Parameters(
      index = "1",
      paramLabel = "ORDERS",
      description = "orders. The format is \"<Item ID>:<Count>,<Item ID>:<Count>,...\"")
  private String orders;

  @Override
  public Integer call() throws Exception {
    String[] split = orders.split(",", -1);
    int[] itemIds = new int[split.length];
    int[] itemCounts = new int[split.length];

    for (int i = 0; i < split.length; i++) {
      String[] s = split[i].split(":", -1);
      itemIds[i] = Integer.parseInt(s[0]);
      itemCounts[i] = Integer.parseInt(s[1]);
    }

    try (Sample sample = new Sample()) {
      System.out.println(sample.placeOrder(customerId, itemIds, itemCounts));
    }

    return 0;
  }
}

File: scalardb-dotnet-samples/scalardb-cluster-linq-sample/Commands/PlaceOrderCommand.cs
using System.CommandLine;

namespace ScalarDbClusterLinqSample.Commands;

public static class PlaceOrderCommand
{
    private const string Name = "PlaceOrder";
    private const string Description = "Place an order";

    private const string ArgName1 = "customer_id";
    private const string ArgDescription1 = "customer ID";
    private const string ArgName2 = "orders";
    private const string ArgDescription2 = "orders. The format is \"<Item ID>:<Count>,<Item ID>:<Count>,...\"";

    public static Command Create()
    {
        var customerIdArg = new Argument<int>(ArgName1, ArgDescription1);
        var ordersArg = new Argument<Dictionary<int, int>>(
            name: ArgName2,
            parse: arg =>
            {
                var argStr = arg.Tokens.First().Value;
                var orders = argStr
                             .Split(',')
                             .Select(s => s.Split(':'))
                             .ToDictionary(
                                 s => Int32.Parse(s[0]),
                                 s => Int32.Parse(s[1])
                             );

                return orders;
            },
            description: ArgDescription2)
        {
            Arity = ArgumentArity.ExactlyOne
        };

        var placeOrderCommand = new Command(Name, Description)
        {
            customerIdArg,
            ordersArg
        };

        placeOrderCommand.SetHandler(async (customerId, orders) =>
        {
            using var sample = new Sample();
            var order = await sample.PlaceOrder(customerId, orders);
            
            Console.WriteLine(order);
        }, customerIdArg, ordersArg);

        return placeOrderCommand;
    }
}
